ocaml/xapi/xapi_vm_group_helpers.ml

```diff
@@ -181,18 +184,24 @@ let update_alert ~__context ~groups ~action =
   with e -> error "%s" (Printexc.to_string e)
 
 let update_vm_anti_affinity_alert ~__context ~groups =
-  update_alert ~__context ~groups
-    ~action:update_vm_anti_affinity_alert_for_group
+  if Pool_features.is_enabled ~__context Features.VM_group then
+    update_alert ~__context ~groups
+      ~action:update_vm_anti_affinity_alert_for_group
+  else
+    debug "VM group feature is disabled, alert will not be updated"
 
 let remove_vm_anti_affinity_alert ~__context ~groups =
-  update_alert ~__context ~groups
-    ~action:remove_vm_anti_affinity_alert_for_group
+  if Pool_features.is_enabled ~__context Features.VM_group then
```

Comment:
The `remove_vm_anti_affinity_alert` is called in `VM_group.destroy`. I understand this is allowed in any case.